code/parachain/frame/ibc/README.md

```diff
@@ -0,0 +1,525 @@
+### Pallet IBC
+
+Pallet IBC is a thin wrapper around [`ibc-rs`](/code/centauri/ibc) that satisfies the runtime requirements for communicating over the IBC protocol from a substrate runtime.
+
+- [`Config`](/code/parachain/frame/ibc/src/lib.rs#L204)
+- [`Call`](/code/parachain/frame/ibc/src/lib.rs#L533)
+- [`Pallet`](/code/parachain/frame/ibc/src/lib.rs#L268)
+
+### Dispatchable functions
+
+- `deliver` - Receives a batch of ibc messages and processes them
+- `transfer` - This initiates an ics20 token transfer from the caller to an account on a connected chain via the ICS20 application
+- `set_params` - Sets parameters that determine whether token transfer or receipt is allowed in ICS20
+- `upgrade_client` - Sets the new consensus state and client state for client upgrades to be executed on connected chains
+
+### Adding Ibc to a substrate runtime
+
+Implementing the ibc config trait for a substrate runtime
+```rust
+parameter_types! {
+	pub const ExpectedBlockTime: u64 = 6000;
+	pub const RelayChainId: light_client_commomn::RelayChain = light_client_commomn::RelayChain::Rococo;
+}
+
+impl pallet_ibc::Config for Runtime {
+	type TimeProvider = Timestamp;
+	type Event = Event;
+	type Currency = Balances;
+	const INDEXING_PREFIX: &'static [u8] = b"ibc/";
+	const CONNECTION_PREFIX: &'static [u8] = b"ibc/";
+	const CHILD_TRIE_KEY: &'static [u8] = b"ibc/";
```

Comment:
I wonder if it makes sense to refactor pallet ibc itself to just use one param? and use them for all 3

Reply:
Yeah that makes sense.